backend/src/posts/depends.py

```diff
@@ -31,47 +31,18 @@ def get_article_upvote_details(article_id, user_id, database):
 
 # get news' information according to search_term, and return the data which the function found.
 def get_new_info(search_term, is_initial=False):
-    all_news_data = []
     # iterate pages to get more news data, not actually get all news data
+    # return crawler.get_headline(search_term, (1, 10) if is_initial else 1)
     if is_initial:
-        news = []
-        for page in range(1, 10):
-            pageinfo = {
-                "page": page,
-                "id": f"search:{quote(search_term)}",
-                "channelId": 2,
-                "type": "searchword",
-            }
-            response = requests.get(Constants.NEWS_LINK, params=pageinfo)
-            news.append(response.json()["lists"])
-
-        for result in news:
-            all_news_data.append(result)
+        all_news_data = crawler.startup(search_term=search_term)
     else:
-        pageinfo = {
-            "page": 1,
-            "id": f"search:{quote(search_term)}",
-            "channelId": 2,
-            "type": "searchword",
-        }
-        response = requests.get(Constants.NEWS_LINK, params=pageinfo)
-        all_news_data = response.json()["lists"]
+        all_news_data = crawler.get_headline(search_term,page=Constants.INIT_PAGE_NUM)
```

Comment:
The line 39 **search the first page** for the news headlines by the given search term, and **it's different from the initial search** that fetches 10 pages of results. As a results, the name `INIT_PAGE_NUM` is a little ambiguous.